controllers/TuitController.ts

```diff
@@ -214,59 +212,24 @@ export default class TuitController implements TuitControllerI {
-        const oldTags = oldT2Ts.map((oldT2T: { tag: any; }) => oldT2T.tag); // Turn T2T array into tag array
-        const oldTagTexts = oldTags.map((oldTag) => oldTag.tag);
-        // Check if the tag is no longer present in req.body
-        for (let i = 0; i < oldTags.length; i++) {
-            if (tuitText.includes("#" + oldTags[i].tag) == 0) {
-                // If not, then remove the Tuit2Tag row
-                await TuitController.tuit2TagDao.deleteTuit2Tag(req.params.tid, oldTags[i]._id);
-                // If it was the last Tuit with that Tag,
-                if (oldTags[i].count === 1) {
-                    // then delete the Tag
-                    await TuitController.tagDao.deleteTag(oldTags[i]._id)
-                } else {
-                    //Reduce count
-                    oldTags[i].count--;
-                }
-            }
-        }
+        // Retrieve a copy of the existing tuit to be updated from database
+        const tuitToBeUpdated = await TuitController.tuitDao.findTuitById(req.params.tid);
 
-        // Check if a new Tag is present
-        // Initialize variables
-        const splitTuit = tuitText.split(" ");
-        let almostTag, newTag;
-
-        // Check if Tuit text contains a tag
-        if (tuitText.includes('#')) {
-            // Loop through words
-            for (let i = 0; i < splitTuit.length; i++) {
-                // If the first char is #
-                if (splitTuit[i].charAt(0) === '#') {
-                    // Prep a Tag (use the word w/o the #)
-                    almostTag = {
-                        "tag": splitTuit[i].slice(1),
-                        "count": 1
-                    }
-                    // If the almost tag is a new one
-                    if (oldTagTexts.includes(almostTag.tag)===false) {
-                        // Create the tag
-                        await TuitController.tagDao.createTag(almostTag);
-                        // Find the newly created Tag
-                        newTag = await TuitController.tagDao.findTagByText(almostTag.tag);
-                        // and make an entry in Tuit2Tag
-                        await TuitController.tuit2TagDao.createTuit2Tag(req.params.tid, newTag._id);
-                    }
-                }
-            }
-        }
-        // Always update the Tuit
-        TuitController.tuitDao.updateTuit(req.params.tid, req.body)
-            .then(status => res.json(status))
+        // Turn tuit into array of one tuit to be given to helper function
+        const tuits2BeUpdated = [tuitToBeUpdated];
+
+        // Remove old tuit2tags and tags corresponding to given tuit
+        await TuitController.tuitService.deleteTuit2TagsAndUpdateTagsByTuits(tuits2BeUpdated);
+
+        // Update the tuit
+        await TuitController.tuitDao.updateTuit(req.params.tid, req.body)
+            .then(status => res.json(status));
```

Comment:
Does it matter if we have code below the res?  I'm just not sure still if this is acting as a "return" type situation.